src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -191,4 +201,176 @@ export class CircleEditorDocument extends Disposable implements vscode.CustomDoc
       }
     };
   }
+
+  /**
+   * edit _model state when user modified model through json editor
+   */
+  editJsonModel(message: any) {
```

Comment:
Since `message.data` is only usage for `message`, how about fixing it?

```suggestion
  editJsonModel(newModelString: string) {
```

Reply:
We will fix this with change argument in function onMessage in CircleEditorProvider.ts
It will be from document.editJsonModel(message) to document.editJsonModel(message.data)